api/app/Http/Controllers/BaseController.php

```diff
@@ -153,68 +157,4 @@ public function deleteMany(Request $request)
 
         return response(null, 204);
     }
```

Comment:
I can share my own general thought on the subject. I don't believe in over engineering for the sake of it, so I usually have a strategy like this where I start simple and then add complexity when needed. The controller I consider to be used for receiving data from the route, delegate the data to the appropriate location in the application for processing, get the result back and return it to the client. That makes it easy to test and mock, and the controller is not responsible for application logic.

The repository pattern goes nicely with this, as the repositories are then responsible for the logic how to handle the data. If one anticipates that the data storage might change in the future, I'd go with a repository interface and work from there. If I'm quite sure it will not change, I'd implement the repositories as needed.  If a data model is a small sub model of another model, I just let the repository handling the parent model take care of that sub set too. If it's major models, injecting respective repository when needed would then do the trick.

So that's a short summing up of my thinking.